repository/companyRepository.ts

```
@@ -0,0 +1,78 @@
+import { Company } from "../models";
+import { CompanyParams } from "../utils/types";
+
+export class CompanyRepository {
+    async findAll() {
+        try {
+            return await Company.find();
+        } catch (err) {
+            console.log(err);
+            throw new Error("Error in finding companies");
+        }
+    }
+    async findByName(companyName: string) {
+        try {
+            return await Company.findOne({
+                where: {
+                    name: companyName
+                }
+            });
+        } catch (err) {
+            console.log(err);
+            throw new Error("Error in finding company with this name");
+        }
+    }
+    async findById(companyId: number) {
+        try {
+            return await Company.findOne({
+                where: {
+                    id: companyId
+                }
+            });
+        } catch (err) {
+            console.log(err);
+            throw new Error("Error in finding company with this id");
+        }
+    }
+    async create(companyParams: any) {
+        try {
+            return await Company.create({...companyParams}).save();
+        } catch (err) {
+            console.log(err);
+            throw new Error("Error in creating company");
+        }
+    }
+    async edit(companyId: number, companyParams: CompanyParams) {
+        try {
+            const company = await Company.findOne({
+                where: {
+                    id: companyId
+                }
+            });
+            if (!company) {
+                return;
```

Why are you just returning here? If we are trying to edit a company that does not exist we need to throw an error.